integreat_cms/cms/forms/custom_content_model_form.py

```diff
@@ -121,6 +122,19 @@ def clean_content(self):
                 "Replaced %r tag with p tag: %r", tag_type, tostring(monospaced)
             )
 
+        # Set link-external as class for external links
+        for link in content.iter("a"):
+            if href := link.get("href"):
+                if "link-external" not in link.get("class", "").split(" ") and not any(
+                    url in href for url in settings.INTERNAL_URLS
+                ):
```

Comment:
> `https://some.blog/2023/10/31-my-cool-cdn.de-spooky-content-at-a-distance`

As far as I understand it, this would be pointing to a file `https://some.blog/2023/10/31-my-cool-cdn` of type `de-spooky-content-at-a-distance`, which is not really likely

> or we can apply `re.sub(r"https?://(www)?", "", …) `to both `url` and `href` before checking with `href.startswith(url)`.

Would this not make it more likely that `url` is in `href`? Since now you've stripped `https://www.` from `url`, and `integreat.app` is way more likely to occur in the path of an URL than `https://www.integreat.app` (for example), I presume.

Also the substitution does not catch URLs which start with `www.` but have no protocol specified.

I think it would make more sense to strip everything from `href` that is not protocol, `www.`, domain name, or the TLD before performing the same check as currently implemented. However this would probably introduce far more edge cases then are currently likely to happen.

Reply:
> > or we can apply `re.sub(r"https?://(www)?", "", …) `to both `url` and `href` before checking with `href.startswith(url)`.
> 
> Would this not make it more likely that `url` is in `href`? Since now you've stripped `https://www.` from `url`, and `integreat.app` is way more likely to occur in the path of an URL than `https://www.integreat.app` (for example), I presume.

Yes this would make it more likely to occur later in the string. But now we can stop searching for it in the whole url and instead just consider the start of it, since the start of it has to be the true domain of the target.

> Also the substitution does not catch URLs which start with `www.` but have no protocol specified.

True. We could account for that (although I think URLs should always have the protocol specified).

> I think it would make more sense to strip everything from `href` that is not protocol, `www.`, domain name, or the TLD before performing the same check as currently implemented. However this would probably introduce far more edge cases then are currently likely to happen.

If we try stripping the items in a strict order this should not be much of a problem, right? E.g.:

```python
base_url = (href
  .removeprefix("http://")
  .removeprefix("https://")
  .removeprefix("www."))
# repeat the same for `url`
```
and also do `base_url = base_url.split("/")[0]` while we're at it to really limit it to the domain.

…this might actually be the best option (besides parsing the URL of course), huh…

Reply:
Possibly better to be iterated over / discussed as a separate issue